### PR TITLE
Compare flag2

### PR DESCRIPTION
**Pull request by amitgilad3**

### CODE REVIEW

File: e2e/commands/show.e2e.js [hunk 1]
```diff
@@ -360,4 +360,17 @@ function add(a, b) {
...
+    beforeEach(() => {
+      helper.initNewLocalScope();
+      helper.createComponentBarFoo();
+      helper.createComponent('bar', 'index.js');
+      helper.addComponentWithOptions('bar/', { i: 'bar/foo' });
+    });
+
+    it('Should throw error that component is not in module', () => {
+      const showCmd = () => helper.showComponent('bar/foo --compare');
+      expect(showCmd).to.throw('error - Unable to compare bar/foo, component not in modules');
```
Comment (GiladShoham) [inline e2e/commands/show.e2e.js]: change error - nothing to compare no previous versions found

File: e2e/commands/show.e2e.js [hunk 2]
```diff
@@ -360,4 +360,17 @@ function add(a, b) {
...
   });
+  describe('with compare flag', () => {
+    beforeEach(() => {
+      helper.initNewLocalScope();
+      helper.createComponentBarFoo();
+      helper.createComponent('bar', 'index.js');
+      helper.addComponentWithOptions('bar/', { i: 'bar/foo' });
+    });
+
+    it('Should throw error that component is not in module', () => {
```
Comment (GiladShoham) [inline e2e/commands/show.e2e.js]: change description according to the new error below

File: src/api/consumer/lib/get-consumer-component.js [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
-export default function getConsumerBit({ id }: { id: string }) {
-  return loadConsumer().then(consumer => consumer.loadComponent(BitId.parse(id)));
-}
+export default (async function getConsumerBit({ id, compare }: { id: string, compare: boolean }) {
+  const consumer = await loadConsumer();
+  const bitId = BitId.parse(id);
+  const component = await consumer.loadComponent(bitId); // loads recent component
+  if (compare) {
+    try {
+      const moduleComponent = await consumer.scope.loadComponent(bitId);
```
Comment (GiladShoham) [inline src/api/consumer/lib/get-consumer-component.js]: change var to componentModel

File: src/cli/commands/public-cmds/show-cmd.js [hunk 4]
```diff
@@ -12,14 +12,18 @@ export default class Show extends Command {
   alias = '';
   opts = [
     ['j', 'json', 'return a json version of the component'],
-    ['v', 'versions', 'return a json of all the versions of the component']
+    ['v', 'versions', 'return a json of all the versions of the component'],
+    ['c', 'compare [boolean]', 'compare current file system component to latest commited component [default=latest]']
```
Comment (GiladShoham) [inline src/cli/commands/public-cmds/show-cmd.js]: change committed to tagged

File: src/cli/commands/public-cmds/show-cmd.js [hunk 5]
```diff
@@ -30,19 +34,22 @@ export default class Show extends Command {
       }));
     }
 
-    return getBitComponent().then(component => ({
+    return getBitComponent().then(({ component, moduleComponent }) => ({
```
Comment (GiladShoham) [inline src/cli/commands/public-cmds/show-cmd.js]: change to componentModel

File: src/cli/default-error-handler.js [hunk 6]
```diff
@@ -143,7 +145,8 @@ const errorsMap: [[Error, (err: Error) => string]] = [
     invalidIdOnCommit,
     err => `error - Unable to tag. ${chalk.bold(err.id)} not found.
 Run \`bit status\` command to list all components available for tag.`
-  ]
+  ],
+  [NothingToCompareTo, err => 'change error - nothing to compare no previous versions found']
```
Comment (GiladShoham) [inline src/cli/default-error-handler.js]: what is this change error?

Comment (amitgilad3) [reply src/cli/default-error-handler.js]: @GiladShoham - fixed typo

**Pull request merged**